guides/github.md

```diff
@@ -0,0 +1,130 @@
+# How to secure GitHub repositories
+
+It is recommended to turn on 2FA for most repositories, especially if building
+a package that is a dependency of multiple other packages.
+
+At least for critical packages, administrators should enable branch
+protection. Require CI to pass before merging to the main branch. Allow only
```

Comment:
I don't understand how branch protection improves *security*. Attacks on github repos have been carried out previously through forks. If the package is on hackage, it is easy to recover the code.

Reply:
> The xz attack was possible because Jia Tan was able to write to the repository directly

That's a matter of access, not of branch protection rules.

You can have secure access policies without any branch protection rules.

I'd even argue it's detrimental to security if you:

- protect master branch
- give all sorts of people write access to non-protected branches 

It's preferable to interact with contributors through forks.

Reply:
(sorry for coming late to this, was swamped by other tasks at work and continuity tasks in face of layoffs)

I partially agree with you. Yes, interacting with contributors via forks is ideal. But having branch protection also protects against accidental pushes to master. By branch protection here all I mean is that merging to master should be done by approved users, only after CI passes, from PRs and not direct pushes.

I'll make a new PR to expand on this